packages/concerto-vocabulary/lib/vocabulary.js

```diff
@@ -145,15 +145,42 @@ class Vocabulary {
      */
     validate(modelFile) {
         const getOwnProperties = (d) => {
-            // ensures we have a valid return, even for scalars
-            return d.getOwnProperties?.() ? d.getOwnProperties?.() : [];
+            // ensures we have a valid return, even for scalars and map-declarations
+            return d.isMapDeclaration() ? [d.getKey(), d.getValue()] :  d.getOwnProperties?.() ? d.getOwnProperties?.() : [];
+        };
+
+        const getPropertyName = (p) => {
+            if(p.isKey?.()) {
+                return 'KEY';
+            } else if(p.isValue?.()) {
+                return 'VALUE';
+            } else {
+                return p.getName();
+            }
+        };
+
+        const checkProperties = (k, p) => {
+            const declaration = modelFile.getLocalType(Object.keys(k)[0]);
+            const property = Object.keys(p)[0];
+            if(declaration.isMapDeclaration()) {
+                if (!property.localeCompare('KEY')) {
```

Comment:
Better to call out the implicit comparison.
```suggestion
                if (property.localeCompare('KEY') === 0) {
```

Reply:
Also, do you need to use `localeCompare`?

Reply:
Now using `===`. Resolved.